src/pages/workspace/WorkspaceInvitePage.js

```diff
@@ -65,93 +64,71 @@ const defaultProps = {
     ...policyDefaultProps,
 };
 
-class WorkspaceInvitePage extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.inviteUser = this.inviteUser.bind(this);
-        this.clearErrors = this.clearErrors.bind(this);
-        this.getExcludedUsers = this.getExcludedUsers.bind(this);
-        this.toggleOption = this.toggleOption.bind(this);
-        this.updateOptionsWithSearchTerm = this.updateOptionsWithSearchTerm.bind(this);
-        this.openPrivacyURL = this.openPrivacyURL.bind(this);
-
-        const {personalDetails, userToInvite} = OptionsListUtils.getMemberInviteOptions(props.personalDetails, props.betas, '', this.getExcludedUsers());
-        this.state = {
-            searchTerm: '',
-            personalDetails,
-            selectedOptions: [],
-            userToInvite,
-        };
-    }
-
-    componentDidMount() {
-        this.clearErrors();
-        const policyMemberEmailsToAccountIDs = PolicyUtils.getClientPolicyMemberEmailsToAccountIDs(this.props.policyMembers, this.props.personalDetails);
-        Policy.openWorkspaceInvitePage(this.props.route.params.policyID, _.keys(policyMemberEmailsToAccountIDs));
-    }
-
-    componentDidUpdate(prevProps) {
-        if (!_.isEqual(prevProps.personalDetails, this.props.personalDetails) || !_.isEqual(prevProps.policyMembers, this.props.policyMembers)) {
-            this.updateOptionsWithSearchTerm(this.state.searchTerm);
-        }
-
-        const isReconnecting = prevProps.network.isOffline && !this.props.network.isOffline;
-        if (!isReconnecting) {
+function WorkspaceInvitePage(props) {
+    const [searchTerm, setSearchTerm] = useState('');
+    const [selectedOptions, setSelectedOptions] = useState([]);
+    const [personalDetails, setPersonalDetails] = useState([]);
+    const [userToInvite, setUserToInvite] = useState(null);
+    const openWorkspaceInvitePageRef = useRef(null);
+
+    openWorkspaceInvitePageRef.current = () => {
+        const policyMemberEmailsToAccountIDs = PolicyUtils.getClientPolicyMemberEmailsToAccountIDs(props.policyMembers, props.personalDetails);
+        Policy.openWorkspaceInvitePage(props.route.params.policyID, _.keys(policyMemberEmailsToAccountIDs));
+    };
```

Comment:
Not sure why need a ref for this, Odd. We could go with the normal function or could use `useCallback` hook?

Reply:
This function should be called when mounting or offline status changes to online.

For doing this, we can use one dependency like `[props.network.isOffline]`. but in this case, the latest value of other dependencies like `props.policyMembers` won't be used due to the nature of Javascript closure.

To use the latest value, we need to add dependencies, but then this functions would be called when other dependencies  changes. This isn't expected. 

In order to use one dependency(`props.network.isOffline`) and use the latest values of dependencies, we need to use ref. Although ref value isn't changed, its `current` property will use the latest values. This is one of common practices for handling these cases.

Reply:
How are you going to do after using `useCallback`? Will you add the function to the dependency list or not? There exists issue for both cases as I explained above

Reply:
Yeah will add a dependency list here for a callback, that would be `props.policyMembers`, `props.personalDetails`,
`props.route.params.policyID`